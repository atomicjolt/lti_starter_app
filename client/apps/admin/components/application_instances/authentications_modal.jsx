import _ from 'lodash';
import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ReactModal from 'react-modal';

import { checkApplicationInstanceAuth } from '../../actions/application_instances';

const select = state => ({
  authenticationChecks: state.authenticationChecks
});

export class AuthenticationsModal extends React.Component {
  static propTypes = {
    isOpen: PropTypes.bool.isRequired,
    closeModal: PropTypes.func.isRequired,
    authentications: PropTypes.array,
    checkApplicationInstanceAuth: PropTypes.func.isRequired,
    applicationInstance: PropTypes.shape({
      id: PropTypes.number,
      config: PropTypes.string,
      site: PropTypes.shape({
        id: PropTypes.number,
      })
    }),
    application: PropTypes.shape({
      id: PropTypes.number,
    }),
    authenticationChecks: PropTypes.shape({}),
  };

  showInstanceModal() {
    if (this.props.isOpen) {
      return 'is-open';
    }
    return '';
  }

  closeModal() {
    this.props.closeModal();
  }

  checkAuth(authentication) {
    this.props.checkApplicationInstanceAuth(
      this.props.application.id,
      this.props.applicationInstance.id,
      authentication.id,
    );
  }

  renderAccounts(authenticationId) {
    const accounts = this.props.authenticationChecks[authenticationId];
    return _.map(accounts, account => (
      <tr key={`auth_${authenticationId}_${account.id}`}>
        <td />
        <td>Can Access:</td>
        <td>{account.name}</td>
        <td>Id: {account.id}</td>
        <td>ParentId: {account.parent_account_id}</td>
        <td>RootId: {account.root_account_id}</td>
      </tr>
    ));
  }

  renderRows() {
    return _.map(this.props.authentications, authentication => (
      <tbody key={`auth_${authentication.id}`}>
        <tr>
          <td>{authentication.id}</td>
          <td>{authentication.user}</td>
          <td>{authentication.email}</td>
          <td>{authentication.provider}</td>
          <td>{authentication.created_at}</td>
          <td>
            <button
              className="c-btn c-btn--gray"
              onClick={() => this.checkAuth(authentication)}
            >
              Check
            </button>
          </td>
        </tr>
        {this.renderAccounts(authentication.id)}
      </tbody>
    ));
  }

  render() {
    const title = 'Authentications';

    return (
      <ReactModal
        isOpen={this.props.isOpen}
        onRequestClose={() => this.closeModal()}
        contentLabel="Application Instances Modal"
        overlayClassName="c-modal__background"
        className={`c-modal c-modal--settings ${this.showInstanceModal()}`}
      >
        <div className="o-grid o-grid__modal-top">
          <h2 className="c-modal__title">
            {title}
          </h2>
          <table className="c-table c-table--instances">
            <thead>
              <tr>
                <th><span>id</span></th>
                <th><span>User</span></th>
                <th><span>Email</span></th>
                <th><span>Provider</span></th>
                <th><span>CREATED</span></th>
                <th />
              </tr>
            </thead>
            { this.renderRows() }
          </table>
        </div>
        <button
          type="button"
          className="c-btn c-btn--gray--large u-m-right"
          onClick={() => this.props.closeModal()}
        >
          Close
        </button>
      </ReactModal>
    );
  }
}

export default connect(select, { checkApplicationInstanceAuth })(AuthenticationsModal);

