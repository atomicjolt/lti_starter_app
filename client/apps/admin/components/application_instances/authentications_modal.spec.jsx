import React from 'react';
import TestRenderer from 'react-test-renderer';

import { authentications_modal } from './authentications_modal';

describe('authentications_modal', () => {
  it('renders the authentications_modal component', () => {
    const result = TestRenderer.create(
        <authentications_modal />
    );
    expect(result).toMatchSnapshot();
  });
});
