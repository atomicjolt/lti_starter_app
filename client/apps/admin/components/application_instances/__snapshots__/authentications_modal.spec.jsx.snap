// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authentications_modal renders the authentications_modal component 1`] = `<authentications_modal />`;
